shopelectro/tests/tests_views.py

```diff
@@ -877,28 +876,29 @@ class InPack(ViewsTestCase):
 
     fixtures = ['dump.json']
 
-    def pack_product(self):
-        product = models.Product.objects.first()
-        product.in_pack = 2
-        product.save()
-        return product
+    def test_catalog_in_pack_units(self):
+        category = (
+            models.Product.objects
+            .annotate(category_size=Count('category__products'))
+            .filter(in_pack__gt=1, category_size__lt=settings.PRODUCTS_ON_PAGE_PC)
+            .first()
+            .category
+        )
 
-    def assert_in_pack_units(self, soup, product):
+        soup = self.get_category_soup(category)
         results = soup.find_all(string=re.compile('упаковка'))
 
-        self.assertEqual(len(results), 1, results)
-        self.assertIn(str(int(product.price)), results[0], results[0])
-
-    def test_catalog_in_pack_units(self):
-        product = self.pack_product()
-        self.assert_in_pack_units(
-            self.get_category_soup(product.category),
-            product,
+        self.assertEqual(
+            len(results),
+            category.products.filter(in_pack__gt=1).count(),
+            results,
         )
 
     def test_product_in_pack_units(self):
-        product = self.pack_product()
-        self.assert_in_pack_units(
-            self.get_product_soup(product),
-            product,
-        )
+        product = models.Product.objects.filter(in_pack__gt=1).first()
+
+        soup = self.get_product_soup(product)
+        results = soup.find_all(string=re.compile('упаковка'))
+
+        self.assertEqual(len(results), 1, results)
+        self.assertIn(str(int(product.price)), results[0], results[0])
```

Comment:
i suggested it on another PR, but missed your answer. It seems it should work too
```suggestion
        self.assertIn(str(product.price), results[0], results[0])
```

Reply:
@duker33 It doesn't work, because we render prices without floating point part:
`self.assertIn(str(100.0), '100 руб / упаковка')`